### PR TITLE
fix: improve error message when importing bad snapshot

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- improve error message when importing bad snapshot
- add unit tests to cover bad snapshot file path, bad snapshot  file, bad snapshot url cases
- (off topic) move `lru` dep to workspace manifest, remove stale comments

After fix:
```
 ...
 2022-10-03T07:10:47.153Z ERROR forest::daemon                 > Failed miserably while importing chain from snapshot Cargo.toml: Failed to parse CAR file: Serialization error for Cbor protocol: Mismatch { expect_major: 5, byte: 112 }
 2022-10-03T07:10:47.153Z ERROR forest::cli                    > Error: Database version 1 is too old (minimum 16 expected). Download a snapshot from a trusted source and import with --import-snapshot=[file|url]
```

```
 ...
 2022-10-03T07:14:28.541Z ERROR forest::daemon                 > Failed miserably while importing chain from snapshot dummy.car: could not open `dummy.car`
 2022-10-03T07:14:28.541Z INFO  forest_rpc                     > Ready for RPC connections
 2022-10-03T07:14:28.541Z ERROR forest::cli                    > Error: Database version 1 is too old (minimum 16 expected). Download a snapshot from a trusted source and import with --import-snapshot=[file|url]
```

**Reference issue to close (if applicable)**

Closes https://github.com/ChainSafe/forest/issues/1974

**Other information and links**

### CODE REVIEW

File: forest/src/daemon.rs [hunk 1]
```diff
@@ -436,28 +442,52 @@ mod test {
     use fvm_shared::address::Address;
 
     #[async_std::test]
-    async fn import_snapshot_from_file() {
+    async fn import_snapshot_from_file_valid() -> anyhow::Result<()> {
+        anyhow::ensure!(import_snapshot_from_file("test_files/chain4.car")
+            .await
+            .is_ok());
+        Ok(())
```
Comment (LesnyRumcajs) [inline forest/src/daemon.rs:449]: Are there some hidden features of this approach as opposed to a classical assertion without `anyhow::Result` as the return type?
```
async fn import_snapshot_from_file_invalid() {
  assert!(import_snapshot_from_file("test_files/chain4.car").await.is_ok())
}
```

Comment (hanabi1224) [reply forest/src/daemon.rs:449]: The error message will be more descriptive, including the failing expression
`import_snapshot_from_file("test_files/chain4.car")
            .await
            .is_ok()`

Comment (LesnyRumcajs) [reply forest/src/daemon.rs:449]: Alright, sounds good.

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]
Thank you. May Forest one day be free of `unwrap` and `expect`. :)

**Pull request merged**